ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -1166,7 +1174,11 @@ data ChainSyncClientException =
       -- | The upstream node's chain contained a block that we know is invalid.
     | forall blk. LedgerSupportsProtocol blk =>
         InvalidBlock
-          (Point blk)  -- ^ Invalid block
+          (Point blk)
+          -- ^ Block that triggered the validity check.
+          (HeaderHash blk)
+          -- ^ Invalid block. If it was pipelined, this can be different from
```

Comment:
Reminder about updating "If it was pipelined": this can happen even if it wasn't pipelined, as long as pipelining is enabled on the peer-to-peer connection.